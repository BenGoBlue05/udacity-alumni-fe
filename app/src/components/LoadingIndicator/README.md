## LoadingIndicator Component
A component that shows as a loading indicator while loading.

### Example

```js
<LoadingIndicator isLoading message="Submitting" />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **isLoading**    | Boolean   | True            | True or false.
| **message**    | String   | "Loading"            | Any string value.
