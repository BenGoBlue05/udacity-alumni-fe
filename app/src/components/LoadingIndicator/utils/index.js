// helper to get random message
export function getRandomMessage(messages) {
  return messages[Math.floor(Math.random() * messages.length)];
}
