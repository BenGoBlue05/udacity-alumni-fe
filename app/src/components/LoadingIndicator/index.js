import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Spinning from 'grommet-udacity/components/icons/Spinning';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';
import messageData from './data/messageData';
import { getRandomMessage } from './utils';

const LoadingIndicator = ({
  isLoading,
  message,
}) => (
  <Box
    align="center"
    justify="center"
    className={styles.loadingIndicator}
  >
    {isLoading &&
      <Box
        align="center"
        justify="center"
      >
        <Spinning />
        <Heading
          tag="h3"
          align="center"
          style={{ maxWidth: 200 }}
        >
          {message}
        </Heading>
      </Box>
    }
  </Box>
);

LoadingIndicator.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  message: PropTypes.string,
};

LoadingIndicator.defaultProps = {
  isLoading: true,
  message: getRandomMessage(messageData),
};

export default cssModules(LoadingIndicator, styles);
