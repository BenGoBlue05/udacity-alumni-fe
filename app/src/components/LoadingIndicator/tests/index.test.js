import LoadingIndicator from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<LoadingIndicator />', () => {
  it('should render with default props', () => {
  	const isLoading = true;
  	const message = 'Do several jumping jacks while you wait...';
    const wrapper = shallow(
      <LoadingIndicator isLoading={isLoading} message={message} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
