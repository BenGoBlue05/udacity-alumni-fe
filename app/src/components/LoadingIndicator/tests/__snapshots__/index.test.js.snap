exports[`<LoadingIndicator /> should render with default props 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  justify="center"
  pad="none"
  responsive={true}
  tag="div">
  <Box
    align="center"
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="none"
    responsive={true}
    tag="div">
    <Spinning />
    <Heading
      align="center"
      style={
        Object {
          "maxWidth": 200
        }
      }
      tag="h3">
      Contemplating banana bread recipes...
    </Heading>
  </Box>
</Box>
`;
